gef.py

```diff
@@ -2172,7 +2172,12 @@ def read_cstring_from_memory(address, max_length=GEF_MAX_STRING_LENGTH, encoding
 
     char_t = cached_lookup_type("char")
     char_ptr = char_t.pointer()
-    res = gdb.Value(address).cast(char_ptr).string(encoding=encoding).strip()
+    try:
+        res = gdb.Value(address).cast(char_ptr).string(encoding=encoding).strip()
+    except gdb.error:
+        length = min(address|(DEFAULT_PAGE_SIZE-1), max_length+1)
```

Comment:
why is it `max_length+1`?

Reply:
Is it for the null byte? I don't think it's necessary

Reply:
In case there is max_length characters, you will like to display it instead of append `...` after it. So you need to check whether the max_length+1 character is null or not.

Reply:
Ah!